src/index.ts

```diff
@@ -199,6 +200,17 @@ class Envoy {
         this.anonymous_agents[agent_id] = hha_hash;
       }
 
+      // Signal is a message initiated in conductor which is sent to UI. In case to be able to route signals
+      // to appropriate agents UIs we need to be able to identify connection based on agent_id and hha_hash.
+
+      // make sure dna2hha entry exists for given hha
+      this.recordHha(hha_hash);
+      let signal_id = this.createSignalId(agent_id, hha_hash);
+
+      // create namespace with signal_id and subscribe this client to this namespace
+      // TODO: Am I subscribing this connection only to this namespace?
+      this.ws_server.event('signal', signal_id);
```

Comment:
I am not sure if this line is doing what I am hoping it does.

My intention here is that:
- on the connection I create an event with a given `signal_id` namespace, but only for THIS particular connection
- once emitting an event in `signal_id` namespace I am hoping it is emitted only to a pipe that opened connection above.

Can someone help me understand if that's how namespaces in [rpc-websocket](https://github.com/elpheria/rpc-websockets/blob/1ba79751f3c0635e9f90f2badf5559355cfe94c1/API.md#serverofname---namespace) work?

Reply:
Looking at the documentation it looks like the way that a client enters a namespace is by being connected to a particular path. For example `this.ws_server.event('signal', '/holo/agent/signup')` would create an event that only clients connected to `/holo/agent/signup` would receive. https://github.com/elpheria/rpc-websockets/blob/1ba79751f3c0635e9f90f2badf5559355cfe94c1/API.md#serverofname---namespace

Reply:
Yes, this has been changed in next commit, because what we really need here is to create an identifiable event that chaperone can sign up to.